### ISSUE TITLE
App freeze on start, sometimes

### ISSUE THREAD

**Issue by valscion** [liballeg/allegro5#798]
I wanted to split this particular failure out from #796 so I could shed some more insight and logs particulary for the case when I:

1. Start the `speed` demo app
2. See the display being generated
3. Display never gets any contents and the application (but not the entire machine) freezes

<img width="649" alt="screen shot 2017-08-11 at 22 45 13" src="https://user-images.githubusercontent.com/482561/29229427-013ddad8-7ee7-11e7-848c-f4312c2fc324.png">

In this case, also the title of the window seems to not be retina ready, which is strange.

---

I checked system logs, and saw this being generated in all sys logs:

```
default 22:37:22.199734 +0300 kernel  [speed pid:1797 mux-aware] starting, non-mux-aware app count 0, switch-state 0 IG FBs 0 EG FBs 1:f power-state 0 3D busy HDA idle
default 22:37:22.199812 +0300 kernel  [speed pid:1797 non-mux-aware] starting, non-mux-aware app count 1, switch-state 0 IG FBs 0 EG FBs 1:f power-state 0 3D busy HDA idle
default 22:37:22.287670 +0300 kernel  NVDA(Graphics): Channel exception! Exception type = 0x45 Graphics Engine Error (GR Class Error)
default 22:37:22.287680 +0300 kernel  NVDA(Graphics): Channel exception! Exception type = 0x45 Graphics Engine Error (GR Class Error)
default 22:37:22.287681 +0300 kernel  NVDA: Channel Info: [19, 0x13, 0x12, 0x16e]
default 22:37:22.287682 +0300 kernel  Channel Info: [19, 0x13, 0x12, 0x16e]
default 22:37:22.287684 +0300 kernel  NVDA: Version Info: [com.apple.GeForce, 10.1.7, 0x7d780b0a, 18894120, 355.10.05.45f01, 1]
default 22:37:22.287685 +0300 kernel  Version Info: [com.apple.GeForce, 10.1.7, 0x7d780b0a, 18894120, 355.10.05.45f01, 1]
default 22:37:22.287687 +0300 kernel  
default 22:37:22.287687 +0300 kernel  
default 22:37:22.287688 +0300 kernel  NVDA: Resource Manager Info:
default 22:37:22.287689 +0300 kernel  Resource Manager Info:
default 22:37:22.287691 +0300 kernel  NVDA:  4443564e 000000d0 13271b11 b4a91b3b 00000001 00000014 d3793533 46d3a4a6
default 22:37:22.287692 +0300 kernel   4443564e 000000d0 13271b11 b4a91b3b 00000001 00000014 d3793533 46d3a4a6
default 22:37:22.287694 +0300 kernel  NVDA:  4614f297 e71edccf 00088301 0000009c 12f2270a 08220a24 1002c197 18408080
default 22:37:22.287696 +0300 kernel   4614f297 e71edccf 00088301 0000009c 12f2270a 08220a24 1002c197 18408080
default 22:37:22.287697 +0300 kernel  NVDA:  28002000 3800300c 9c40249c 81814824 50088098 60135800 9a030a00 0a0a0014
default 22:37:22.287700 +0300 kernel   28002000 3800300c 9c40249c 81814824 50088098 60135800 9a030a00 0a0a0014
default 22:37:22.287701 +0300 kernel  NVDA:  18071382 90005000 3d0a0101 0a3a1392 00002438 00000e00 00e00100 000f4900
default 22:37:22.287703 +0300 kernel   18071382 90005000 3d0a0101 0a3a1392 00002438 00000e00 00e00100 000f4900
default 22:37:22.287704 +0300 kernel  NVDA:  00000100 000b4900 00000000 03044700 0000ff00 0000ff00 0000ff00 0000ff00
default 22:37:22.287706 +0300 kernel   00000100 000b4900 00000000 03044700 0000ff00 0000ff00 0000ff00 0000ff00
default 22:37:22.287712 +0300 kernel  NVDA:  0000ff00 0000ff00 c2210a00 00081e13 a8180010 0bf7f080 45281320 00380330
default 22:37:22.287713 +0300 kernel   0000ff00 0000ff00 c2210a00 00081e13 a8180010 0bf7f080 45281320 00380330
default 22:37:22.287715 +0300 kernel  NVDA:  9dd9f850 40119293 00480280 4443564e
default 22:37:22.287716 +0300 kernel   9dd9f850 40119293 00480280 4443564e
default 22:37:22.287717 +0300 kernel  
default 22:37:22.287718 +0300 kernel  
default 22:37:22.287718 +0300 kernel  NVDA: Accelerator Event History:
default 22:37:22.287720 +0300 kernel  Accelerator Event History:
default 22:37:22.287721 +0300 kernel  NVDA:  0a0808001a04080010010a0808001a04080210010a2a0800122608c080021080c0879f
default 22:37:22.287722 +0300 kernel   0a0808001a04080010010a0808001a04080210010a2a0800122608c080021080c0879f
default 22:37:22.287724 +0300 kernel  NVDA:  84f0ffffff01188182c08ebc82c00920c7808080a00928ef828080a0020a230800121f
default 22:37:22.287725 +0300 kernel   84f0ffffff01188182c08ebc82c00920c7808080a00928ef828080a0020a230800121f
default 22:37:22.287727 +0300 kernel  NVDA:  08c480021080c0879f84f0ffffff01188182c08ebc82c009208182800828000a080800
default 22:37:22.287729 +0300 kernel   08c480021080c0879f84f0ffffff01188182c08ebc82c009208182800828000a080800
default 22:37:22.287730 +0300 kernel  NVDA:  1a04080210000a0808001a0408001000
default 22:37:22.287732 +0300 kernel   1a04080210000a0808001a0408001000
default 22:37:22.302512 +0300 coreaudiod  HALS_IOContextDescription.cpp:1352:___ZN25HALS_IOContextDescription24PushStreamUsageToEnginesEb_block_invoke:  IOContext 102: pushing 1 input stream(s) to 'AppleHDAEngineInput:1B,0,1,0:1'
default 22:37:22.302547 +0300 coreaudiod  HALS_IOContextDescription.cpp:1356:___ZN25HALS_IOContextDescription24PushStreamUsageToEnginesEb_block_invoke:  IOContext 102: input stream[0] => Enabled: 'AppleHDAEngineInput:1B,0,1,0:1'
default 22:37:22.305815 +0300 coreaudiod  HALS_IOContextDescription.cpp:1352:___ZN25HALS_IOContextDescription24PushStreamUsageToEnginesEb_block_invoke:  IOContext 103: pushing 1 output stream(s) to 'AppleHDAEngineOutput:1B,0,1,1:0'
default 22:37:22.305855 +0300 coreaudiod  HALS_IOContextDescription.cpp:1356:___ZN25HALS_IOContextDescription24PushStreamUsageToEnginesEb_block_invoke:  IOContext 103: output stream[0] => Enabled: 'AppleHDAEngineOutput:1B,0,1,1:0'
default 22:37:22.308197 +0300 kernel  + IOAudioEngine[<private>]::newUserClient(0x<private>, <private>, 0x0, 0x0, <private>)
default 22:37:22.308205 +0300 kernel  + IOAudioEngineUserClient[<private>]::initWithAudioEngine(<private>, 0x<private>, <private>, 0x0, 0x0)
default 22:37:22.308211 +0300 kernel  +- IOAudioEngineUserClient[<private>]::setOnline(0)
default 22:37:22.308215 +0300 kernel    ConnectionID:0xc0a89b59
default 22:37:22.308222 +0300 kernel  - IOAudioEngineUserClient[<private>]::initWithAudioEngine(<private>, 0x<private>, <private>, 0x0, 0x0) returns 1
default 22:37:22.308224 +0300 kernel  +- IOAudioEngineUserClient::withAudioEngine(<private>, 0x<private>, <private>, 0x0, 0x0) returns <private>
default 22:37:22.308230 +0300 kernel  + IOAudioEngine::addUserClientAction(<private>, <private>)
default 22:37:22.308232 +0300 kernel  +- IOAudioEngine[<private>]::addUserClient(<private>) returns 0x0
default 22:37:22.308233 +0300 kernel  - IOAudioEngine::addUserClientAction(<private>, <private>) returns 0x0
default 22:37:22.308236 +0300 kernel  - IOAudioEngine[<private>]::newUserClient(0x<private>, <private>, 0x0, 0x0, <private>)
default 22:37:22.308249 +0300 kernel  - IOAudioEngineUserClient[<private>]::registerFormatNotification(0x<private>, 0x100) returns 0x0
default 22:37:22.308251 +0300 kernel  - IOAudioEngineUserClient::registerNotificationAction(<private>, <private>) returns 0x0
default 22:37:22.308253 +0300 kernel  - IOAudioEngineUserClient[<private>]::registerNotificationPort(0x<private>, 0x0, 0x100) returns 0x0
default 22:37:22.309645 +0300 kernel  + IOAudioEngine[<private>]::newUserClient(0x<private>, <private>, 0x0, 0x0, <private>)
default 22:37:22.309653 +0300 kernel  + IOAudioEngineUserClient[<private>]::initWithAudioEngine(<private>, 0x<private>, <private>, 0x0, 0x0)
default 22:37:22.309657 +0300 kernel  +- IOAudioEngineUserClient[<private>]::setOnline(0)
default 22:37:22.309661 +0300 kernel    ConnectionID:0x52ff9f92
default 22:37:22.309667 +0300 kernel  - IOAudioEngineUserClient[<private>]::initWithAudioEngine(<private>, 0x<private>, <private>, 0x0, 0x0) returns 1
default 22:37:22.309670 +0300 kernel  +- IOAudioEngineUserClient::withAudioEngine(<private>, 0x<private>, <private>, 0x0, 0x0) returns <private>
default 22:37:22.309675 +0300 kernel  + IOAudioEngine::addUserClientAction(<private>, <private>)
default 22:37:22.309678 +0300 kernel  +- IOAudioEngine[<private>]::addUserClient(<private>) returns 0x0
default 22:37:22.309679 +0300 kernel  - IOAudioEngine::addUserClientAction(<private>, <private>) returns 0x0
default 22:37:22.309682 +0300 kernel  - IOAudioEngine[<private>]::newUserClient(0x<private>, <private>, 0x0, 0x0, <private>)
default 22:37:22.309694 +0300 kernel  - IOAudioEngineUserClient[<private>]::registerFormatNotification(0x<private>, 0x101) returns 0x0
default 22:37:22.309696 +0300 kernel  - IOAudioEngineUserClient::registerNotificationAction(<private>, <private>) returns 0x0
default 22:37:22.309698 +0300 kernel  - IOAudioEngineUserClient[<private>]::registerNotificationPort(0x<private>, 0x0, 0x101) returns 0x0
default 22:37:22.312260 +0300 coreaudiod  HALS_IOContextDescription.cpp:1352:___ZN25HALS_IOContextDescription24PushStreamUsageToEnginesEb_block_invoke:  IOContext 103: pushing 1 output stream(s) to 'AppleHDAEngineOutput:1B,0,1,1:0'
default 22:37:22.312291 +0300 coreaudiod  HALS_IOContextDescription.cpp:1356:___ZN25HALS_IOContextDescription24PushStreamUsageToEnginesEb_block_invoke:  IOContext 103: output stream[0] => Enabled: 'AppleHDAEngineOutput:1B,0,1,1:0'
default 22:37:22.329108 +0300 speed 3243: aq@0x7fd5e0052200 volume 1.000 1.000 1.000 1.000 1.000
default 22:37:22.329396 +0300 speed 194: AQME device AQDefaultOutput: client starting: <AudioQueueObject@0x7fd5e0052200; [0]; play>; running count now 1
default 22:37:22.329546 +0300 speed 879: aqmeio@0x7fd5df464e00, device 0x2f (AQDefaultOutput), AudioDeviceStart (err 0)
default 22:37:22.329693 +0300 kernel  + IOAudioEngineUserClient::externalMethod, selector=0x0,   arg0 0x0, arg1 0x107f72000, arg2 0x8010 arg3 0x67
default 22:37:22.329702 +0300 kernel    scalarInputCount=0x4  structureInputSize 0x0, scalarOutputCount 0x0, structureOutputSize 0x0
default 22:37:22.329732 +0300 kernel  + IOAudioEngineUserClient::registerBuffer64 0x0 0x107f72000 0x8010 0x67
default 22:37:22.329736 +0300 kernel  + IOAudioEngineUserClient::safeRegisterClientBuffer64 <private>
default 22:37:22.329741 +0300 kernel  + IOAudioEngineUserClient[<private>]::registerClientBuffer64  (<private>[1], 0x107f72000, 0x8010, 0x67)
default 22:37:22.329743 +0300 kernel    null clientBufferSetList
default 22:37:22.329744 +0300 kernel    did not find clientBufferSetList for ID 0x67
default 22:37:22.329772 +0300 kernel    clientBuffer->mAudioClientBuffer32.sourceBuffer before offset: <private>, offset size: 16
default 22:37:22.329774 +0300 kernel    clientBuffer->mAudioClientBuffer32.sourceBuffer after offset: <private>
default 22:37:22.329775 +0300 kernel    null clientBufferSetList
default 22:37:22.329776 +0300 kernel    did not find clientBufferSetList for ID 0x67
default 22:37:22.329778 +0300 kernel    creating new IOAudioClientBufferSet
default 22:37:22.329780 +0300 kernel    output
default 22:37:22.329781 +0300 kernel    !isOnline
default 22:37:22.329783 +0300 kernel  - IOAudioEngineUserClient::registerClientBuffer64() result 0x0
default 22:37:22.329785 +0300 kernel  - IOAudioEngineUserClient::safeRegisterClientBuffer64  returns 0x0
default 22:37:22.329788 +0300 kernel  - IOAudioEngineUserClient::registerBuffer64 0x0 0x107f72000 0x8010 0x67 returns 0x0
default 22:37:22.329797 +0300 kernel  - IOAudioEngineUserClient::externalMethod returns 0x0
default 22:37:22.329818 +0300 kernel  + IOAudioEngineUserClient::externalMethod, selector=0x3,   arg0 0x0, arg1 0x0, arg2 0x0 arg3 0x0
default 22:37:22.329820 +0300 kernel    scalarInputCount=0x0  structureInputSize 0x0, scalarOutputCount 0x0, structureOutputSize 0x0
default 22:37:22.329840 +0300 kernel  + IOAudioEngineUserClient[<private>]::startClient() - 0
default 22:37:22.329842 +0300 kernel    audioEngine && !isInactive(). State = 0
default 22:37:22.329843 +0300 kernel    audioEngine->getState() != kIOAudioEnginePaused
default 22:37:22.329848 +0300 kernel  +- IOAudioEngineUserClient[<private>]::setOnline(1)
default 22:37:22.329850 +0300 kernel    !isOnline() setting online
default 22:37:22.329853 +0300 kernel  + IOAudioEngine[<private>]::startClient(<private>)
default 22:37:22.329856 +0300 kernel  + IOAudioEngine[<private>]::incrementActiveUserClients() - 0
default 22:37:22.329860 +0300 kernel  + IOAudioEngine[<private>]::startAudioEngine(state = 0)
default 22:37:22.329864 +0300 kernel  + IOAudioDevice[<private>]::protectedCompletePowerStateChange() - current = 2 - pending = 2
default 22:37:22.329867 +0300 kernel  - IOAudioDevice[<private>]::protectedCompletePowerStateChange() - current = 2 - pending = 2 returns 0x0
default 22:37:22.329872 +0300 kernel  + IOAudioDevice[<private>]::protectedCompletePowerStateChange() - current = 2 - pending = 2
default 22:37:22.329874 +0300 kernel  - IOAudioDevice[<private>]::protectedCompletePowerStateChange() - current = 2 - pending = 2 returns 0x0
default 22:37:22.329878 +0300 kernel  + IOAudioDevice[<private>]::audioEngineStarting() - numRunningAudioEngines = 0
default 22:37:22.329880 +0300 kernel  - IOAudioDevice[<private>]::audioEngineStarting() - numRunningAudioEngines = 1
default 22:37:22.329882 +0300 kernel  +- IOAudioEngine[<private>]::resetStatusBuffer()
default 22:37:22.334652 +0300 opendirectoryd  Client: <private>, UID: 0, EUID: 0, GID: 0, EGID: 0
default 22:37:22.341542 +0300 opendirectoryd  Client: <private>, UID: 0, EUID: 0, GID: 0, EGID: 0
default 22:37:22.342018 +0300 kernel  +-IOAudioEngine[<private>]::setState(0x1. oldState=0)
default 22:37:22.342025 +0300 kernel  - IOAudioEngine[<private>]::getTimerInterval()
default 22:37:22.342097 +0300 kernel  - IOAudioEngine[<private>]::startAudioEngine() returns 0x0
default 22:37:22.342099 +0300 kernel  - IOAudioEngine[<private>]::incrementActiveUserClients() - 1 returns 0
default 22:37:22.342102 +0300 kernel  - IOAudioEngine[<private>]::startClient(<private>) returns 0x0
default 22:37:22.342103 +0300 kernel    engine started
default 22:37:22.342105 +0300 kernel    bufferSet <private>
default 22:37:22.342106 +0300 kernel    output clientBuffer <private>
default 22:37:22.342108 +0300 kernel  + IOAudioStream[<private>]::addClient(<private>)
default 22:37:22.342113 +0300 kernel  - IOAudioStream[<private>]::addClient(<private>) returns 0x0
default 22:37:22.342115 +0300 kernel  - IOAudioEngineUserClient[<private>]::startClient() - 1 returns 0x0
default 22:37:22.342118 +0300 kernel  - IOAudioEngineUserClient::externalMethod returns 0x0
default 22:37:22.345178 +0300 DumpGPURestart  Spawning /usr/bin/tailspin (1800) to save GPURestart-triggered tailspin log (<private>)
default 22:37:22.354922 +0300 DumpGPURestart  Saved gpuRestart report for kernel to <private>
default 22:37:22.364336 +0300 DumpGPURestart  Removing excessive log: file:///Library/Logs/DiagnosticReports/Kernel_2017-08-09-215626_vesa-mbp.gpuRestart
default 22:37:22.764358 +0300 opendirectoryd  Client: <private>, UID: 0, EUID: 0, GID: 0, EGID: 0
default 22:37:22.764471 +0300 opendirectoryd  Client: <private>, UID: 0, EUID: 0, GID: 0, EGID: 0
default 22:37:23.264301 +0300 trustd  cert[2]: AnchorTrusted =(leaf)[force]> 0
default 22:37:23.266395 +0300 com.apple.iCloudHelper  MacOS error: -67050
default 22:37:23.457895 +0300 DumpGPURestart  tailspin save exited abnormally 22
default 22:37:23.458634 +0300 kernel  SandboxViolation: DumpGPURestart(1799) deny(1) mach-lookup com.apple.CrashReporterSupportHelper
default 22:37:23.458988 +0300 DumpGPURestart  Connection error while checking Apple Internalness. Error: Error Domain=NSCocoaErrorDomain Code=4099 "The connection to service named com.apple.CrashReporterSupportHelper was invalidated." UserInfo={NSDebugDescription=The connection to service named com.apple.CrashReporterSupportHelper was invalidated.}
```

It tells that more details can be found from a specific log file:

```
default 22:37:22.354922 +0300 DumpGPURestart  Saved gpuRestart report for kernel to <private>
```

The gpuRestart log file was generated successfully.

**Here's the full logs:**: https://gist.github.com/valscion/4ac8cd5c88a63c40856ef0d961a55cf5

This seems to be what happened, as seen from the gpuRestart log:

```
Fri Aug 11 22:37:22 2017

Event:               GPU Reset
Date/Time:           Fri Aug 11 22:37:22 2017
Application:         speed
Path:                
Tailspin:            /Library/Logs/DiagnosticReports/gpuRestart2017-08-11-223722.tailspin
GPUSubmission Trace ID: 0
OS Version:          Mac OS X Version 10.12.6 (Build 16G29)
Graphics Hardware:   NVIDIA GeForce GT 650M
Signature:           45

Report Data:

NVDA(Graphics): Channel exception! Exception type = 0x45 Graphics Engine Error (GR Class Error)
Channel Info: [19, 0x13, 0x12, 0x16e]
Version Info: [com.apple.GeForce, 10.1.7, 0x7d780b0a, 18894120, 355.10.05.45f01, 1]
```

**Comment by valscion** [comment]
I got this again, and now with allegro.log enabled, here's the contents:

https://gist.github.com/valscion/e787514dd0163d7a6f41444723734f4a

**Comment by valscion** [comment]
And again, adding even more logging output for this:

https://gist.github.com/valscion/69eae32756c2d17772753a349a4b4271

**Comment by valscion** [comment]
Actually, it might just be that the logging slowed the starting up down so that the program might actually become responsive again. I'll have to make sure it isn't the case.

**Comment by beoran** [comment]
Could you perhaps compile the examples that come with Allegro 5 sources and
try to run them all, or at least the screen related ones? That could give
us some better idea of where the problem might be.

**Comment by valscion** [comment]
Nah, I'm pretty certain that both of those logs are realistic. I see a gpuRestart log being generated in system logs:

<details>
<summary>First freeze</summary>

```
Fri Aug 11 23:42:08 2017

Event:               GPU Reset
Date/Time:           Fri Aug 11 23:42:08 2017
Application:         speed
Path:                
Tailspin:            /Library/Logs/DiagnosticReports/gpuRestart2017-08-11-234208.tailspin
GPUSubmission Trace ID: 0
OS Version:          Mac OS X Version 10.12.6 (Build 16G29)
Graphics Hardware:   NVIDIA GeForce GT 650M
Signature:           20

Report Data:

NVDA(Graphics): Channel exception! Exception type = 0x20 DMA Engine Error (PBDMA Error)
Channel Info: [16, 0xf, 0xf, 0x2]
Version Info: [com.apple.GeForce, 10.1.7, 0x7d780b0a, 18894120, 355.10.05.45f01, 1]

Resource Manager Info:
 4443564e 000000cc 3f625592 dad03aad 00000001 00000014 d3793533 46d3a4a6
 4614f297 e71edccf 00088301 00000097 12f2220a 081d0a1f 1002c197 20001810
 30002800 05dc3800 4805dc40 00500392 00600f58 149a030a 820a0a00 00180713
 01900050 923d0a01 380a3a13 00000024 0000000e 0000e001 00000f49 00000001
 00000b49 00000000 00030447 000000ff 000000ff 000000ff 000000ff 000000ff
 000000ff 13c2210a 1000081e 80a81800 200bf7fc 3020280f 50003803 c4b09c98
 8040378e 00014802 4443564e

 4443564e 000000cc 51ff59a8 623dad9c 00000001 00000014 d3793533 46d3a4a6
 4614f297 e71edccf 00088301 00000097 12f2220a 081d0a1f 1002c197 20001810
 30002800 05dc3800 4805dc40 00500392 00600f58 149a030a 820a0a00 00180713
 01900050 923d0a01 380a3a13 00000024 0000000e 0000e001 00000f49 00000001
 00000b49 00000000 00030447 000000ff 000000ff 000000ff 000000ff 000000ff
 000000ff 13c2210a 1000081e 80a81800 200bf7fc 3020280f 50003803 c4b585b0
 8040378e 00024802 4443564e

 4443564e 00000104 f3cfcad3 e3f43138 00000001 00000014 d3793533 46d3a4a6
 4614f297 e71edccf 00088301 000000cd 12ea2d0a 080c0a2a 180f1000 bfae2000
 2a0786fd 1001080a 02808288 2a010122 1001080e 02818080 80820522 0a088080
 2612f229 0008240a 80808010 20001801 30002800 18843800 80988440 fd480880
 02808887 0f580050 030a0060 0a00149a 0713820a 00500018 0a010190 3a13923d
 0024380a 000e0000 e0010000 0f490000 00010000 0b490000 00000000 04470000
 00ff0003 00ff0000 00ff0000 00ff0000 00ff0000 00ff0000 210a0000 081e13c2
 18001000 f7fc80a8 280f200b 3803300d 9bd05000 378ec4b8 48028040 00000003
 4443564e

Accelerator Event History:
 0a0808001a04080010010a0808001a04080210010a2a0800122608c08002108080e7cd
 83f0ffffff0118e181c08e8c82c00720c680808090092883808080f0010a230800121f
 08c48002108080e7cd83f0ffffff0118e181c08e8c82c007208182800828000a080800
 1a04080210000a0808001a0408001000
```

</details>

<details>
<summary>Second freeze</summary>

```
Fri Aug 11 23:47:21 2017

Event:               GPU Reset
Date/Time:           Fri Aug 11 23:47:21 2017
Application:         speed
Path:                
Tailspin:            /Library/Logs/DiagnosticReports/gpuRestart2017-08-11-234721.tailspin
GPUSubmission Trace ID: 0
OS Version:          Mac OS X Version 10.12.6 (Build 16G29)
Graphics Hardware:   NVIDIA GeForce GT 650M
Signature:           20

Report Data:

NVDA(Graphics): Channel exception! Exception type = 0x20 DMA Engine Error (PBDMA Error)
Channel Info: [16, 0xf, 0xf, 0x25]
Version Info: [com.apple.GeForce, 10.1.7, 0x7d780b0a, 18894120, 355.10.05.45f01, 1]

Resource Manager Info:
 4443564e 000000cc fa25a426 23e49afd 00000001 00000014 d3793533 46d3a4a6
 4614f297 e71edccf 00088301 00000097 12f2220a 081d0a1f 1002c197 20001810
 30002800 05dc3800 4805dc40 00500392 00600f58 149a030a 820a0a00 00180713
 01900050 923d0a01 380a3a13 00000024 0000000e 0000e001 00000f49 00000001
 00000b49 00000000 00030447 000000ff 000000ff 000000ff 000000ff 000000ff
 000000ff 13c2210a 1000081e 80a81800 200bf7fc 3020280f 50003803 c1dde5c8
 8040409a 00044802 4443564e

 4443564e 00000100 62c1ae47 10e75dfc 00000001 00000014 d3793533 46d3a4a6
 4614f297 e71edccf 00088301 000000ca 12ea2d0a 080c0a2a 180f1000 90e22001
 2a089bfc 1001080a 02808288 2a010122 1001080e 02818080 80820522 0a088080
 2312f226 f408210a 8080100c 00180180 00280020 b8380030 a6b84026 48088080
 58005000 0a00600f 00149a03 13820a0a 50001807 01019000 13923d0a 24380a3a
 0e000000 01000000 490000e0 0100000f 49000000 0000000b 47000000 ff000304
 ff000000 ff000000 ff000000 ff000000 ff000000 0a000000 1e13c221 00100008
 fc80a818 0f200bf7 03300d28 e8500038 9ac1e4ef 02804040 00000548 4443564e

Accelerator Event History:
 0a0808001a04080010010a0808001a04080210010a2a0800122608c080021080e0c2c5
 83f0ffffff0118f181c08e8c82c00720c5808080800928a6808080f0010a230800121f
 08c480021080e0c2c583f0ffffff0118f181c08e8c82c007208182800828000a080800
 1a04080210000a0808001a0408001000
```

</details>


Soo... a DMA Engine Error?!?

**Comment by valscion** [comment]
> Could you perhaps compile the examples that come with Allegro 5 sources and
> try to run them all, or at least the screen related ones? That could give
> us some better idea of where the problem might be.

Yeah, I'll try that. I'm sorry if I keep spamming your email as I debug this further 😅

**Comment by beoran** [comment]
http://themacblog.com/how-to/fix-kernel-nvdaopengl-channel-exception/

It may be a HW problem, or so I read after googling the error.

As for spamming, could you perhaps wait until you have run all example
programs and collected all logs you think relevant and then post them in
one big message? It a bit hard to follow this thread if you post your
results immediately when you get them.

KR, B.

**Comment by valscion** [comment]
> It may be a HW problem, or so I read after googling the error.

Yeah, I'm afraid that might be the case. However it's strange that I've only ever encountered this with Allegro, and that I can consistently reproduce this on Allegro with a few tries.

> As for spamming, could you perhaps wait until you have run all example
> programs and collected all logs you think relevant and then post them in
> one big message? It a bit hard to follow this thread if you post your
> results immediately when you get them.

The trouble is, I'm not sure which logs are relevant and I'm afraid I'm gonna forget to post some critical piece of information so I just... send all the things once I hit a failure.

I'll try to be a bit more methodological with my logging and group them together.
